zopkio/adhoc_deployer.py

```diff
@@ -110,18 +110,17 @@ def install(self, unique_id, configs=None):
       process = self.processes[unique_id]
       prev_hostname = process.hostname
       if 'hostname' in configs:
-        if prev_hostname is not configs['hostname']:
-          self.uninstall(unique_id, configs)
-          hostname = configs['hostname']
-        else:
-          self.uninstall(unique_id, configs)
-          hostname = prev_hostname
+        self.uninstall(unique_id, configs)
+        hostname = configs['hostname']
     elif 'hostname' in configs:
```

Comment:
this if and elif are the same (not your change but can you clean up). Also if running with test_mode_localhost and hostname is not set we probably don't want to raise a deploymenterror